bin/release.js

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env node
+import {fileURLToPath} from 'url'
+import {dirname, join} from 'pathe'
+import {createRequire} from 'module'
+import {readFileSync} from 'fs'
+
+const require = createRequire(import.meta.url)
+const {Octokit} = require('@octokit/rest')
+const execa = require('execa')
+
+const octokit = new Octokit({
+    auth: process.env.GITHUB_TOKEN,
+});
+const goos = ['linux', 'windows', 'darwin']
+const goarch =  ['386', 'amd64']
+const goPath = process.env.PATHGO
+const baseReleaseParams = {
+    owner: 'shopify',
+    repo: 'cli',
+}
+const binDirectory = dirname(fileURLToPath(import.meta.url))
+const rootDirectory = dirname(binDirectory)
+const extensionsPath = join(rootDirectory, 'packages/ui-extensions-go-cli')
+let cliVersion = getCliVersion()
+
+const versionsMatrix = generateVersionsMatrix()
+await Promise.all(versionsMatrix.map(generateVersion))
+const release = await createRelease()
+await Promise.all(versionsMatrix.map((version) => updloadReleaseAssetVersion(version, release)))
+await Promise.all(versionsMatrix.map(cleanGeneratedVersion))
```

Comment:
Note that if any of these steps fails, the binaries generated by `generateVersion` will remain uncleaned. I'd recommend using the pattern that we've followed throughout the codebase to create a temporary directory and let the API deal with the cleaning:

```ts
inTemporaryDirectory(async (tmpDir) => {
  / .. build logic
  // 🔥 Automatically deleted when the function exits
})
```

Note that you'll have to adjust the `package.js` executable to accept the path where the file will be output:

```js
await execa("yarn", ["package", binaryPath] /*....*/)
```